source/extensions/filters/common/expr/context.cc

```diff
@@ -339,17 +338,11 @@ absl::optional<CelValue> FilterStateWrapper::operator[](CelValue key) const {
     if (cel_state) {
       return cel_state->exprValue(&arena_, false);
     } else if (object != nullptr) {
-      // Attempt to find the reflection object.
-      auto factory =
-          Registry::FactoryRegistry<StreamInfo::FilterState::ObjectFactory>::getFactory(value);
-      if (factory) {
-        auto reflection = factory->reflect(object);
-        if (reflection) {
-          auto* raw_reflection = reflection.release();
-          arena_.Own(raw_reflection);
-          return CelValue::CreateMap(
-              ProtobufWkt::Arena::Create<FilterStateObjectWrapper>(&arena_, raw_reflection));
-        }
+      // TODO(wbpcode): the implementation of cannot handle the case where the object has provided
+      // field support, but callers only want to access the whole object.
+      if (object->hasFieldSupport()) {
+        return CelValue::CreateMap(
+            ProtobufWkt::Arena::Create<FilterStateObjectWrapper>(&arena_, object));
```

Comment:
I think this is a little weird. We always treat the filter state object that has reflection/field support as map. So, we cannot support the case where the object has related support but we want's to access the whole object (by serializeAsString()).

Reply:
Yes, I'm aware of this limitation. It's same with proto Struct.